### ISSUE TITLE
Bug: allowSeparatedGroups in sort-imports rule does nothing

### ISSUE THREAD

**Issue by adrm** [eslint/eslint#17464]
### Environment

Node version: v16.18.1
npm version: v8.19.2
Local ESLint version: Not found
Global ESLint version: Not found
Operating System: linux 6.4.4-200.fc38.x86_64

Per npm ls eslint: eslint@8.30.0

### What parser are you using?

@typescript-eslint/parser

### What did you do?

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgc29ydC1pbXBvcnRzOiBbXCJlcnJvclwiLCB7IFwiYWxsb3dTZXBhcmF0ZWRHcm91cHNcIjogdHJ1ZSB9XSovXG5cbmltcG9ydCBiIGZyb20gJ2Zvby5qcyc7XG5cbmltcG9ydCBjIGZyb20gJ2Jhci5qcyc7XG5pbXBvcnQgYSBmcm9tICdiYXouanMnO1xuIiwib3B0aW9ucyI6eyJlbnYiOnt9LCJydWxlcyI6eyJzb3J0LWltcG9ydHMiOlsiZXJyb3IiLHsiYWxsb3dTZXBhcmF0ZWRHcm91cHMiOmZhbHNlfV19LCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e30sImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6Im1vZHVsZSJ9fX0=

### What did you expect to happen?

That code, for the config enabled, should give two errors: one because imports aren't sorted and other because imports are in separated groups. 

### What actually happened?

The sort-imports rule is triggered, because we get one error, but not for checking the grouped imports case, since the error is not reflected even though the example is copied from an example error from documentation.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgc29ydC1pbXBvcnRzOiBbXCJlcnJvclwiLCB7IFwiYWxsb3dTZXBhcmF0ZWRHcm91cHNcIjogdHJ1ZSB9XSovXG5cbmltcG9ydCBiIGZyb20gJ2Zvby5qcyc7XG5cbmltcG9ydCBjIGZyb20gJ2Jhci5qcyc7XG5pbXBvcnQgYSBmcm9tICdiYXouanMnO1xuIiwib3B0aW9ucyI6eyJlbnYiOnt9LCJydWxlcyI6eyJzb3J0LWltcG9ydHMiOlsiZXJyb3IiLHsiYWxsb3dTZXBhcmF0ZWRHcm91cHMiOmZhbHNlfV19LCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e30sImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6Im1vZHVsZSJ9fX0=

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by Rec0iL99** [comment]
Hi @adrm, thanks for the issue. This is not a bug. 

```js
/* eslint sort-imports: ["error", { "allowSeparatedGroups": true }] */
```

Setting the `allowSeparatedGroups: true` option will allow you to create individual group imports so that sort order in each group will be evaluated separately. By default `allowSeparatedGroups` is set to `false` in the `sort-imports` rule so all imports are grouped into one and evaluated.

```js
/* eslint sort-imports: ["error", { "allowSeparatedGroups": true }] */

// group 1
import b from 'foo.js';

// group 2
import c from 'bar.js';
import a from 'baz.js';
```

The rule only reports one error because group 2 imports are unsorted. Group 1 is sorted.

The correct fix:

```js
/* eslint sort-imports: ["error", { "allowSeparatedGroups": true }] */

// group 1
import b from 'foo.js';

// group 2
import a from 'baz.js';
import c from 'bar.js';
```

**Comment by adrm** [comment]
Oh, the comment in the code might have confused you, but please look at the config in that playground. The allowSeparatedGroups setting is set to false:

![imagen](https://github.com/eslint/eslint/assets/307945/c0562e87-2ac8-4e55-b432-e7dc3a9ce1f9)

That comment comes from the docs, since I created the snippet from there. 

The error is not being detected even though allowSeparatedGroups is set to false.

Could you confirm and reopen the issue? Sorry for the confusion and thank you very much.

**Comment by Rec0iL99** [comment]
The rule configuration comment in code will take precedence over the rule configuration in your `.eslintrc.*`. Check out this [example](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IFwibm8tY29uc29sZVwiOiBcIm9mZlwiICovXG5jb25zb2xlLmxvZyhcImhlbGxvIHdvcmxkXCIpIiwib3B0aW9ucyI6eyJlbnYiOnt9LCJydWxlcyI6eyJuby1jb25zb2xlIjpbImVycm9yIl19LCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e30sImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6Im1vZHVsZSJ9fX0=) that demonstrates this. This was what happened in the example you provided.

Even if `allowSeparatedGroups` is `false`, there will be one error only ([example](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgc29ydC1pbXBvcnRzOiBbXCJlcnJvclwiLCB7IFwiYWxsb3dTZXBhcmF0ZWRHcm91cHNcIjogZmFsc2UgfV0qL1xuXG5pbXBvcnQgYiBmcm9tICdmb28uanMnO1xuXG5pbXBvcnQgYyBmcm9tICdiYXIuanMnO1xuaW1wb3J0IGEgZnJvbSAnYmF6LmpzJzsgLy8gZmxhZ2dlZCBiZWNhdXNlICdhJyBpcyBpbiB0aGUgd3JvbmcgcG9zaXRpb25cblxuIiwib3B0aW9ucyI6eyJlbnYiOnt9LCJydWxlcyI6e30sInBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fSwiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoibW9kdWxlIn19fQ==)). If you meant that having separated group imports should be flagged, then I don't think it should because having separated group imports is valid JavaScript.

**Comment by adrm** [comment]
Thanks for the explanation. I now understand that allowSeparatedGroups only decides if the order should be global for all imports on the file or just local for each group of imports. I wrongly understood that allowSeparatedGroups: false would error when making a separated group of imports, regardless of order.

Sorry for misunderstanding!
